hibernate-core/src/main/java/org/hibernate/engine/spi/SessionFactoryImplementor.java

```diff
@@ -299,7 +302,14 @@ default Dialect getDialect() {
 		return getJdbcServices().getDialect();
 	}
 
-	SqlStringGenerationContext getSqlStringGenerationContext();
+	default SqlStringGenerationContext getSqlStringGenerationContext() {
```

Comment:
Why do you need this default implementation? Do you implement this interface in Blaze persistence?!

Because this implementation is not completely correct, so we cannot rely on it in the long run. Unless there's a specific reason beyond "it's an SPI and we broke it", I wouldn't add this default implementation.